### DISCUSSION TITLE
Add Application Version into

### DISCUSSION THREAD

**Discussion by stevenbaert** [Romanitho/Winget-Install-GUI#60] [Ideas]
Might be nice to add also application version into search result of apps? That way user can see version which he's installing.
Sometimes user might need specific version (testing) and then you might offer different versions to install too.

That should be added somwewhere here in your code:

 $index = $lines[$fl] -split '\s+'

    # Line $fl has the header, we can find char where we find ID and Version
    $idStart = $lines[$fl].IndexOf($index[1])
    $versionStart = $lines[$fl].IndexOf($index[2])

    # Now cycle in real package and split accordingly
    $searchList = @()
    For ($i = $fl + 2; $i -le $lines.Length; $i++) {
        $line = $lines[$i]
        if ($line.Length -gt ($sourceStart + 5)) {
            $software = [Software]::new()
            $software.Name = $line.Substring(0, $idStart).TrimEnd()
            $software.Id = $line.Substring($idStart, $versionStart - $idStart).TrimEnd()
            #add formated soft to list
            $searchList += $software
        }
    }
    return $searchList

**Comment by Romanitho** [comment]
Hi,  
You can put arguments (like the version `-v`) in the select app fields, then add the app in your list:
![image](https://user-images.githubusercontent.com/96626929/180763258-c4d10851-0607-4d56-8b18-3050de0e3801.png)

**Reply by stevenbaert** [reply]
Thanks, but that's not what I meant. This will enable you to install a particular version indeed. I meant to show a table in which you can see the values Winget offers you = Vendor, Version, Description, Installer etc. Which is a far more rich experience then just the name.
You could make use of https://www.powershellgallery.com/packages/WinGet/0.0.1 to accomplish this too, but then you are dependant on another PS Module.

**Reply by Romanitho** [reply]
I understand your request, but there are some other tools for that, even online tools ([winget.run](https://winget.run/) or [winstall.app](https://winstall.app/) for instance)
We wanted to offer a simple experience that can help using our Install and AutoUpdate scripts basically.

Edit: we expect a lot from Winget to support powershell natively, that will obviously much simplify this kind of request :)
